vg/vg.go

```diff
@@ -85,6 +88,49 @@ type CanvasSizer interface {
 	Size() (x, y Length)
 }
 
+// CanvasWriterTo wraps behavior required for writing
+// a Canvas representation to a concrete image.
+type CanvasWriterTo interface {
+	CanvasSizer
+	io.WriterTo
+}
+
+var (
+	formatLock sync.RWMutex
+	formats    = make(map[string]func(w, h Length) CanvasWriterTo)
```

Comment:
The thinking that I had while doing this was that we should support easy use of common formats. I'm not completely convinced that this includes OpenGL or X11. I'm troubled by how I'm dealing with EPS, but not hugely. Even if we add something like functional options, we still need to make some formats special - we need to assert on the returned Canvas or make decisions on the basis of the format string (and then provide information that we don't actually have under the current model - Plot.WriterTo would need to be extended to take the functional options to get this).

This needs some thought.